### PR TITLE
Spider rework

### PR DESCRIPTION
**Pull request by hellmonk3**
Adds several new monster types and adjust some old monsters to improve the diversity and playfeel of Spider. This rework attempts to maintain some of the distinct playfeel of spider (getting swarmed by dudes, strategic damage and damage over time effects) while adding a bit more variety, (hopefully) making some of the things that are strong there a little bit less so, and keeping the branch thematically distinct from other sbranches. Currently adds 8 new monsters and partially reworks several others, completely overhauls the spawn table, and modifies bands; after feedback, some new monsters will probably be cut and vaults updated to use the survivors.

### CODE REVIEW

File: crawl-ref/source/dat/descript/monsters.txt
Comment (PleasingFungus) [inline crawl-ref/source/dat/descript/monsters.txt:837]: FWIW, I sort of like the current name.

File: crawl-ref/source/describe.cc
Comment (PleasingFungus) [inline crawl-ref/source/describe.cc:3782]: Flavor: would this be more fun as 'spawn a spider', or would it be too misleading re: summoned status?

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:2125]: Wonder if we should rename this to `MST_HOWLER` or `MST_SCREAMER` or something.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:2149]: Wow, that's one big bite! Vaguely wonder about tuning it down slightly, maybe to 40?

File: crawl-ref/source/areas.cc
Comment (PleasingFungus) [inline crawl-ref/source/areas.cc:615]: Sort of funny to have a halo monster in the ghost moth branch. Very antisynergy. (Also a little antisynergistic with EV monsters generally.)

As long as it's still a dangerous monster, though, that's probably fine..? Just don't want folks luring this around.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:2811]: ...roaches have claws?

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:2814]: Feels slightly mean to have the irradiate monster be fast.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:2864]: This is a bold genus choice.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:3528]: Great theme.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:3536]: Why SIZE_LARGE?

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-data.h:3532]: Sort of funny that these have the exact same attack as hornets, but I suppose there's no overlap in range.

File: crawl-ref/source/mon-death.cc
Comment (PleasingFungus) [inline crawl-ref/source/mon-death.cc:1043]: This is an interesting effect. It says "kill me last". For that to be an interesting tension, rather than "oh, kill this unthreatening monster last", pharaoh ants need to have a strong attack and poor defenses. (Thus, normally something you'd *want* to kill first.)

I think they might have a bit too much HP and a bit too little damage for that at present. What do you think?

File: crawl-ref/source/mon-death.cc
Comment (PleasingFungus) [inline crawl-ref/source/mon-death.cc:1056]: Let's only print this message if something gets bound.

```suggestion
    bool bound = false;
    for (monster_near_iterator mi(mons, LOS_NO_TRANS); mi; ++mi)
    {
        if (*mi == mons || !mons_can_bind_soul(mons, *mi))
            continue;
        if (!bound)
            simple_monster_message(*mons, " binds the souls of nearby monsters.");
        bound = true;
        mi->add_ench(
            mon_enchant(ENCH_BOUND_SOUL, 0, mons,
                        random_range(10, 30) * BASELINE_DELAY));
    }
```

File: crawl-ref/source/mon-death.cc
Comment (PleasingFungus) [inline crawl-ref/source/mon-death.cc:1048]: Do we need this check? Surely pharaoh ants can't bind themselves regardless, since they're already undead, right?

File: crawl-ref/source/mon-death.cc
Comment (PleasingFungus) [inline crawl-ref/source/mon-death.cc:2320]: These `{` `}` look like a bug. Do you really want `_pharaoh_ant_bind_souls` to only trigger if the pharaoh ant itself has ENCH_BOUND_SOUL?

File: crawl-ref/source/mon-gear.cc
Comment (PleasingFungus) [inline crawl-ref/source/mon-gear.cc:517]: One of the 'things' in Spider is that it has no item-using monsters, aside from visiting uniques. It seems easy enough to keep this intact with Jorogumos - since they're spider spirits merely pretending to be human, it makes some sense that they wouldn't be weapon using. (Likewise, they probably shouldn't have robes or `MONUSE_WEAPONS_ARMOUR`.)

If this is important to the design, please push back and let me know.

File: crawl-ref/source/mon-pick-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-pick-data.h:396]: Nice structure.

File: crawl-ref/source/mon-pick-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-pick-data.h:413]: Even with friends, I'm not sure that torpor snails are above the bottom-tier of threats by Spider. I'd vaguely lean toward swapping them with tarantellas in the tier listing. Interested in your thoughts, though.

File: crawl-ref/source/mon-pick-data.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-pick-data.h:441]: Spider doesn't actually have OOD monsters, does it? See 7b68f31d6f572688753d6cbcb57fac3b77f5d3b7 for an example of how to enable OODs, if you want them.

Also feels a bit sad that almost all of these are just high-end 'normal' spawns. Suppose there's not really any super late game stuff to steal, though. Should've added weird hell bugs...

File: crawl-ref/source/mon-spell.h
Comment (PleasingFungus) [inline crawl-ref/source/mon-spell.h:682]: Lol at this plus the AF. Hope the player's near some shallow water :)

File: crawl-ref/source/monster-type.h
Comment (PleasingFungus) [inline crawl-ref/source/monster-type.h:196]: In principle, you should try to sort these into appropriate areas. (Jorogumo near harpy, etc.) It doesn't really matter (for several reasons), though.

File: crawl-ref/source/monster.cc
Comment (PleasingFungus) [inline crawl-ref/source/monster.cc:6347]: The comment at the top of this function needs to be updated. (Probably just deleting the 'currently' section.)

File: crawl-ref/source/monster.cc
Comment (PleasingFungus) [inline crawl-ref/source/monster.cc:6358]: This is awful. Can we add a new flag for this?

I recommend wandering into `mon-flags.h` and adding `M_WEB_IMMUNE = BIT(15)` on line 59, adding it to *at least* the monsters explicitly listed by type here, and then checking it here with `mons_class_flag(type, M_WEB_IMMUNE)`. As a more advanced move, also add it to the non-dummy genus monsters listed here (`MONS_JELLY` etc) and also check `|| mons_class_flag(mons_genus(type), M_WEB_IMMUNE)`.

If you'd rather not do this, please remind me when you merge this branch and I'll make the change myself.

File: crawl-ref/source/rltiles/dc-mon.txt
Comment (PleasingFungus) [inline crawl-ref/source/rltiles/dc-mon.txt:142]: Think most of these (excepting broodmother and maybe steelbarb) should have `%rim 0` set, since they already have black borders drawn on.

File: crawl-ref/source/spl-summoning.cc
Comment (PleasingFungus) [inline crawl-ref/source/spl-summoning.cc:3086]: Minor: think we'd more usually call this just `summon_spider`. We don't usually bother with articles in our function names.

File: crawl-ref/source/util/monster/monster-main.cc
Comment (PleasingFungus) [inline crawl-ref/source/util/monster/monster-main.cc:1061]: IMO this is funnier as just `(spider)`.

**Review by PleasingFungus — APPROVED** [review]
Looks mostly good. Please take a look through my comments (esp re: pharaoh ant & jorōgumo) before merging.

**Review by ebering — APPROVED** [review]
I went commit by commit, ended up commenting on some things that were later tweaked.

Overall impression of the read through: mash that `git rebase --interactive` history revision then merge (or just mash the mf merge button but this history's a bit of a mess).

I started a char I plan to use for playtesting but haven't had the time to give this a good playthrough, so trust your gut and follow your dreams.

File: crawl-ref/source/mon-pick-data.h
Comment (PleasingFungus) [reply crawl-ref/source/mon-pick-data.h:441]: Sure, agree that shouldn't block merge. I was just rambling.

File: crawl-ref/source/mon-data.h
Comment (PleasingFungus) [reply crawl-ref/source/mon-data.h:2811]: Fair.

File: crawl-ref/source/mon-data.h
Comment (hellmonk3) [reply crawl-ref/source/mon-data.h:2809]: Was going to fix this, but funnily enough they're already silver vuln and zin-hated without a flag (due to having a chaotic ability).

**Pull request merged**
